### PR TITLE
Fix UpdateContract()

### LINKED ISSUE #1911: Price for OpCode.SYSCALL can be zero
**Summary or problem description**
In Func `ApplicationEngine.UpdateContract`

https://github.com/neo-project/neo/blob/4066a4f3986a71612dcdeca33a4ecf8934511247/src/neo/SmartContract/ApplicationEngine.Contract.cs#L58

Here input of func `AddGas` can be zero, when both script and manifest is null or with 0 length.

Such script and manifest is also unreasonable.

**Do you have any solution you want to propose?**
Suggest change: throw exception for such script or manifest like here:

https://github.com/neo-project/neo/blob/4066a4f3986a71612dcdeca33a4ecf8934511247/src/neo/SmartContract/ApplicationEngine.Contract.cs#L33

**Neo Version**
- Neo 3

**Where in the software does this update applies to?**
- VM

### PR DESCRIPTION
**Pull request by erikzhang**
Fix #1911

### CODE REVIEW

**Comment by erikzhang** [comment]
@Qiao-Jin

File: src/neo/SmartContract/ApplicationEngine.Contract.cs [hunk 1]
```diff
@@ -55,6 +55,8 @@ protected internal ContractState CreateContract(byte[] script, byte[] manifest)
 
         protected internal void UpdateContract(byte[] script, byte[] manifest)
         {
+            if (script is null && manifest is null) throw new ArgumentException();
```
Comment (Qiao-Jin) [inline src/neo/SmartContract/ApplicationEngine.Contract.cs:58]: Not only null, we should also consider situation when length = 0

Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Contract.cs:58]: The `length` cannot be `0` in `UpdateContract()`.

https://github.com/neo-project/neo/blob/4066a4f3986a71612dcdeca33a4ecf8934511247/src/neo/SmartContract/ApplicationEngine.Contract.cs#L65-L66

https://github.com/neo-project/neo/blob/4066a4f3986a71612dcdeca33a4ecf8934511247/src/neo/SmartContract/ApplicationEngine.Contract.cs#L82-L83

Comment (Qiao-Jin) [reply src/neo/SmartContract/ApplicationEngine.Contract.cs:58]: Maybe we can put the logic into a single line

Comment (erikzhang) [reply src/neo/SmartContract/ApplicationEngine.Contract.cs:58]: No need.

**Review by shargon — APPROVED** [review]

**Pull request merged**